File: main/src/main/java/net/fwitz/math/main/binary/complex/functions/trig/SinZ2.java
package net.fwitz.math.main.binary.complex.functions.trig;

import net.fwitz.math.plot.binary.complex.ComplexFunctionPlot;

public class SinZ2 {
    public static void main(String[] args) {
        new ComplexFunctionPlot("sin(z^2)")
                .computeFn(z -> z.times(z).sin())
                .domainX(-Math.PI, Math.PI)
                .domainY(-Math.PI, Math.PI)
                .render();
    }
}

File: main/src/main/kotlin/net/fwitz/math/main/binary/complex/functions/trig/SinZ2.kt
package net.fwitz.math.main.binary.complex.functions.trig

import net.fwitz.math.plot.binary.complex.ComplexFunctionPlot

object SinZ2 {
    @JvmStatic
    fun main(args: Array<String>) = ComplexFunctionPlot("sin(z^2)")
        .computeFn { z -> z.pow2.sin }
        .domainX(-Math.PI, Math.PI)
        .domainY(-Math.PI, Math.PI)
        .render()
}